File: sol/2327/2327.py
class Solution:
    def peopleAwareOfSecret(self, n: int, delay: int, forget: int) -> int:
        m = 10 ** 9 + 7
        memo = [0] * forget
        memo[0] = 1
        idx = 0
        total = 0
        for _ in range(n - 1):
            idx = (idx + forget - 1) % forget
            total = (total + memo[(idx + delay) % forget] - memo[idx]) % m
            memo[idx] = total
        return sum(memo) % m

File: sol/2327/2327.cpp
struct Solution {
    int peopleAwareOfSecret(int n, int delay, int forget) {
        constexpr int m = 1e9 + 7;
        vector<int> memo(forget, 0);
        memo[0] = 1;
        int idx = 0, total = 0;
        while (n-- > 1){
            idx = (idx + forget - 1) % forget;
            total = (total + memo[(idx + delay) % forget]) % m;
            total = (total + (m - memo[idx])) % m;
            memo[idx] = total;
        }
        total = 0;
        for (int x : memo){
            total = (total + x) % m;
        }
        return total;
    }
};
